icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -5323,4 +5324,33 @@ void RBBITest::TestTraceCreateBreakEngine(void) {
 }
 #endif
 
+void RBBITest::TestUnpairedSurrogate() {
+    UnicodeString rules(u"ab;");
+
+    UErrorCode status = U_ZERO_ERROR;
+    UParseError pe;
+    RuleBasedBreakIterator bi1(rules, pe, status);
+    assertSuccess(WHERE, status);
+    UnicodeString rtRules = bi1.getRules();
+    // make sure the simple one work first.
+    assertEquals(WHERE, UnicodeString(u"ab;"),  rtRules);
+
+
+    UnicodeString rules2 = UnicodeString(u"a\\ud800b;").unescape();
+    RuleBasedBreakIterator bi2(rules2, pe, status);
+    assertEquals(WHERE "unpaired lead surrogate", U_ILLEGAL_CHAR_FOUND , status);
+
+    status = U_ZERO_ERROR;
+    UnicodeString rules3 = UnicodeString(u"a\\ude00b;").unescape();
+    RuleBasedBreakIterator bi3(rules3, pe, status);
+    assertEquals(WHERE "unpaired tail surrogate", U_ILLEGAL_CHAR_FOUND , status);
+
+    // make sure the surrogate one work too.
+    status = U_ZERO_ERROR;
+    UnicodeString rules4 = UnicodeString(u"a\\ud800\\ude00b;").unescape();
```

Comment:
(optional) This could be simplified to
      `UnicodeString rules4(u"a😀b;");`
(The double backslash hex + unescape is only needed for getting unpaired surrogates)